src/Callback.php

```diff
@@ -138,17 +138,25 @@ public function getPaymentStatus(): string
      */
     public function getPaymentId(): string
     {
-        return $this->getValueByName('id', $this->getPayment());
+        return $this->getValue('payment.id');
     }
 
     /**
      * Get signature
      *
      * @return string
+     * @throws ProcessException
      */
     public function getSignature(): string
     {
-        return $this->getValueByName('signature');
+        $signature = $this->getValue('signature')
+            ?? $this->getValue('general.signature');
+
+        if (!$signature) {
```

Comment:
Несколько раз, мне кажется, мы уже обжигались:

```suggestion
        if ($signature === null) {
```

Reply:
Да, ты прав 👍 . Не везде поставил корректную проверку